vault/external_tests/identity/login_mfa_okta_test.go

```diff
@@ -288,22 +299,57 @@ func mfaGenerateOktaLoginMFATest(client *api.Client) error {
 	}
 	mfaConstraints, ok := secret.Auth.MFARequirement.MFAConstraints["randomName"]
 	if !ok {
-		return fmt.Errorf("failed to find the mfaConstrains")
+		return fmt.Errorf("failed to find the mfaConstraints")
 	}
 	if mfaConstraints.Any == nil || len(mfaConstraints.Any) == 0 {
 		return fmt.Errorf("")
 	}
 	for _, mfaAny := range mfaConstraints.Any {
-		if mfaAny.ID != methodID || mfaAny.Type != "okta" {
+		if mfaAny.ID != methodID || mfaAny.Type != "okta" || (mfaAny.UsesPasscode != useTOTP) {
 			return fmt.Errorf("invalid mfa constraints")
 		}
 	}
 
+	// get totp from file if requested by test runner
+	var passcodes []string
+	if useTOTP {
+		// generate tmp file path
+		tempDir := t.TempDir()
+		totpFile := tempDir + string(os.PathSeparator) + "totp.txt"
+
+		t.Logf("Please save your totp to: %s", totpFile)
+
+		// Try to read the file 10x per second or until 5 minutes have passed.
+		timer := time.NewTimer(5 * time.Minute)
+		defer timer.Stop()
+		for {
+			totpFileContents, err := os.ReadFile(totpFile)
+			if err != nil {
+				select {
+				case <-timer.C:
+					break
+				default:
+					time.Sleep(100 * time.Millisecond)
+					continue
+				}
+				// Should this Skip instead of Failing the test due to lack of user interaction?
```

Comment:
This seems fine given that this is only going to be run with a user present and not in CI.